cmd/clairctl/docker/login.go

```diff
@@ -0,0 +1,47 @@
+package docker
```

Comment:
Why don't we simply rely on `docker login` for this and for logout?

For example, the following function provides the required credentials (if the user has used `docker login` on the registry before) for the specified image:
```
// GetAuthCredentials returns the auth credentials (if any found) for the given repository, as found
// in the user's docker config.
func GetAuthCredentials(image string) (types.AuthConfig, error) {
	// Lookup the index information for the name.
	indexInfo, err := registry.ParseSearchIndexInfo(image)
	if err != nil {
		return types.AuthConfig{}, err
	}

	// Retrieve the user's Docker configuration file (if any).
	configFile, err := cliconfig.Load(cliconfig.ConfigDir())
	if err != nil {
		return types.AuthConfig{}, err
	}

	// Resolve the authentication information for the registry specified, via the config file.
	return registry.ResolveAuthConfig(configFile.AuthConfigs, indexInfo), nil
}
```